pkg/sql/create_stats.go

```diff
@@ -104,26 +104,25 @@ type createStatsNode struct {
 // createStatsRun contains the run-time state of createStatsNode during local
 // execution.
 type createStatsRun struct {
-	errCh chan error
+	cancelFn context.CancelFunc
+	errCh    chan error
 }
 
 func (n *createStatsNode) startExec(params runParams) error {
 	telemetry.Inc(sqltelemetry.SchemaChangeCreateCounter("stats"))
 	n.run.errCh = make(chan error)
-	go func() {
-		err := n.startJob(params.ctx)
-		select {
-		case <-params.ctx.Done():
-		case n.run.errCh <- err:
-		}
-		close(n.run.errCh)
-	}()
-	return nil
+	cancelFn, err := n.startJob(params.ctx)
```

Comment:
as per the comment a bottom of this file, if you use a ctxgroup this would change to 

```
	g, err := n.startJob(params.ctx)
	n.run.group = g
	return err
```